judge/judgedaemon.main.php

```diff
@@ -253,16 +251,19 @@ function read_judgehostlog(int $n = 20) : string
 function fetch_executable(
     string $workdirpath, string $type, string $execid, bool $combined_run_compare = false) : array
 {
+    // Safe when you trust this code) (workdirpath+type+execid is safe)
     $execdir = $workdirpath . '/executable/' . $type . '/' . $execid;
     $execdeploypath  = $execdir . '/.deployed';
+    // Safe when execdir is safe
     $execbuilddir    = $execdir . '/build';
     $execbuildpath   = $execbuilddir . '/build';
     $execrunpath     = $execbuilddir . '/run';
     if (!is_dir($execdir) || !file_exists($execdeploypath)) {
-        system("rm -rf $execdir");
-        system("rm -rf $execbuilddir");
-        system("mkdir -p '$execbuilddir'", $retval);
-        if ($retval!=0) {
+        $rm_exec_dir_builddir = escapeshellcmd("rm -rf $execdir $execbuilddir");
```

Comment:
we may want to put the paths here into quotes (see below) since they might contain spaces